docs/ibc-relayer.md

```diff
@@ -0,0 +1,488 @@
+
+## Prerequisites
+
+Before beginning, ensure you have:
+
+1. Rust installed and configured
+2. Access to RPC and gRPC endpoints for both chains
+3. Wallets funded with native tokens for both chains
+
+For this tutorial, we will be using:
+
+- Babylon's `bbn-test-5` testnet
+- Cosmos Hub's `theta-testnet-001` testnet
+
+## Hermes
+
+[Hermes](https://github.com/informalsystems/hermes) is an open-source
+Rust implementation of an IBC relayer released as part of the
+`ibc-relayer-cli` crate. It includes a CLI for relaying packets between
+Cosmos SDK chains, as well as Prometheus metrics and a REST API.
+
+Please follow the steps
+at [Hermes Quick Start](https://hermes.informal.systems/quick-start/)
+to install Hermes. Before proceeding, verify that Hermes is installed correctly
+by
+running `hermes version`.
+
+### Configuration
+
+After you have successfully installed Hermes and created the necessary folders,
+you now have to edit `config.toml` and add the appropriate configurations for
+the chains you want to relay between.
+
+Edit the Hermes configuration:
+
+```bash
+vim $HOME/.hermes/config.toml
+```
+
+Here's the complete configuration for both chains:
+
+```toml
+[global]
+log_level = "info"
+
+[mode.clients]
+enabled = true
+refresh = true
+misbehaviour = true
+
+[mode.connections]
+enabled = false
+
+[mode.channels]
+enabled = false
+
+[mode.packets]
+enabled = true
+clear_interval = 100
+clear_on_start = true
+tx_confirmation = false
+auto_register_counterparty_payee = false
+
+[rest]
+enabled = false
+host = "127.0.0.1"
+port = 3000
+
+[telemetry]
+enabled = false
+host = "127.0.0.1"
+port = 3001
+
+[[chains]]
+id = "theta-testnet-001"
+type = "CosmosSdk"
+rpc_addr = "https://rpc.sentry-01.theta-testnet.polypore.xyz"
+grpc_addr = "https://grpc.sentry-01.theta-testnet.polypore.xyz"
+websocket_addr = "ws://rpc.sentry-01.theta-testnet.polypore.xyz:26657/websocket"
+rpc_timeout = "10s"
+account_prefix = "cosmos"
+key_name = "key-cosmos"
+store_prefix = "ibc"
+default_gas = 100000
+max_gas = 400000
+gas_multiplier = 1.5
+max_msg_num = 30
+max_tx_size = 180000
+clock_drift = "5s"
+max_block_time = "30s"
+trusting_period = "14days"
+client_refresh_rate = "1/3"
+gas_price = { price = 0.025, denom = "uatom" }
+trust_threshold = { numerator = "1", denominator = "3" }
+[chains.packet_filter]
+policy = "allow"
+list = [["transfer", "channel-0"]]
+
+[[chains]]
+id = "bbn-test-5"
+type = "CosmosSdk"
+rpc_addr = "https://babylon-testnet-rpc.nodes.guru"
+grpc_addr = "https://babylon-testnet-grpc.nodes.guru"
+websocket_addr = "wss://babylon-testnet-rpc.nodes.guru/websocket"
+rpc_timeout = "10s"
+account_prefix = "bbn"
+key_name = "babylon-key"
+store_prefix = "ibc"
+default_gas = 100000
+max_gas = 400000
+gas_multiplier = 1.5
+max_msg_num = 30
+max_tx_size = 180000
+clock_drift = "5s"
+max_block_time = "30s"
+# Babylon has a ~50 hour unbonding period, so we set the trusting period to 33 hours
+trusting_period = "33hours"
```

Comment:
Let's have a paragraph (maybe even upfront) on the trusting period setting of Babylon. Imo we need to make the following points clear:

- Babylon uses BTC timestamping assisted unbonding rather than traditional 21-day unbonding in Cosmos SDK
- BTC timestamping assited unbonding time depends on BTC block rate which is variable
- babylon will configure w=300 such that unbonding time will be ~50 hours on average. This leads to trusting period of 33 hours if trusting period = 2/3 unbonding period